src/typing/typer.ml

```diff
@@ -2446,10 +2446,16 @@ and type_call ?(mode=MGet) ctx e el (with_type:WithType.t) inline p =
 		e
 	| (EField(e,"match"),p), [epat] ->
 		let et = type_expr ctx e WithType.value in
-		(match follow et.etype with
-			| TEnum _ ->
-				Matcher.Match.match_expr ctx e [[epat],None,Some (EConst(Ident "true"),p),p] (Some (Some (EConst(Ident "false"),p),p)) (WithType.with_type ctx.t.tbool) true p
-			| _ -> def ())
+		let is_enum_match = match follow et.etype with
+			| TEnum _ -> true
+			| TAbstract _ as t when (match Abstract.follow_with_abstracts_forward t with TEnum _ -> true | _ -> false) ->
+				(try ignore (type_field (TypeFieldConfig.create true) ctx et "match" p mode); false with _ -> true)
```

Comment:
This means that if there's a static extension `match` that works with the abstract, the internal `match` will be hidden. That's fine, but it doesn't work like that for real enums. I wonder if this should be aligned between the two.

Reply:
Yes, knew that. Didn't want to introduce any breaking change. Can do aligned variant.

Reply:
Well, it's not a big deal, just something to keep in mind and maybe handle eventually. I think this change is good regardless.